### DISCUSSION TITLE
Federating timelines and reply threads

### DISCUSSION THREAD

**Discussion by beholdnec** [bluesky-social/atproto#2154] [Clarifications and "Why?"] [answered]
Since federation is becoming more relevant, I figured it was time to ask this.

One problem with Mastodon is that posts and replies might be missing, due to posts not being federated from one instance to another. This can result in a reply thread not being identical from one instance to another. The same is true of user timelines - Mastodon still lacks backfilling as far as I'm aware, so if you browse a user's timeline, it might be missing posts.

What is atproto's story with regard to these issues? I consider this a relatively major problem that could potentially frustrate a lot of people who would otherwise enjoy the service. It was one of the first "well, this sucks" moments I experienced on Mastodon that made me think about switching away.

**Comment by snarfed** [comment, marked as answer]
ATProto generally doesn't suffer from this specific issue because it has a multi-tier architecture. User data servers (PDSes) are expected to be plentiful and widely federated, but their data is then aggregated into other tiers (relays and AppViews) that are more centralized and have much more complete views of the network. Clients generally read that aggregated data from those relays and AppViews, not from individual PDSes.

**Reply by beholdnec** [reply]
Interesting. It seems like we're counting on the relays to have a sufficiently large view of the network.
